core/src/main/java/com/graphhopper/routing/ch/EdgeBasedNodeContractor.java

```diff
@@ -295,9 +298,12 @@ private CHEntry doAddShortcut(CHEntry edgeFrom, CHEntry edgeTo) {
         LOGGER.trace("Adding shortcut from {} to {}, weight: {}, firstOrigEdge: {}, lastOrigEdge: {}",
                 from, adjNode, edgeTo.weight, edgeFrom.getParent().incEdge, edgeTo.incEdge);
         CHEdgeIteratorState shortcut = prepareGraph.shortcut(from, adjNode);
-        long direction = PrepareEncoder.getScFwdDir();
+        int direction = PrepareEncoder.getScFwdDir();
         // we need to set flags first because they overwrite weight etc
-        shortcut.setFlags(direction);
+        // TODO NOW creating a new intsref every time here is probably not efficient ?
+        IntsRef intsRef = new IntsRef(1);
+        intsRef.ints[0] = direction;
+        shortcut.setFlags(intsRef);
         shortcut.setSkippedEdges(edgeFrom.edge, edgeTo.edge)
```

Comment:
Or maybe you have a better idea in this regard?

Reply:
Some possibilities:

 * allow getFlags: this avoids an additional IntsRef object in case of a new shortcut as we can just call "shortcut.getFlags". This is not that bad as getFlags is now also discouraged for edges due to the low level usability
 * introduce accessEnc and weightEnc, which allows e.g. dynamically configuring the precision but the overhead when comparing to pure bit operations might be too high?
 * use a different API for shortcuts, e.g. fallback to `long getShortcutFlags` as everything that we want to store is known before and we could create setter&getter for it (like setWeight or setFirstAndLastOrigEdges). But this is inconsistent compared to normal edges.